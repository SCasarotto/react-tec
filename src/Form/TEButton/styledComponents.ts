import styled from 'styled-components'

export const Button = styled.button`
	position: relative;
	display: block;
	width: 100%;
	font-size: 18px;
	padding: 5px 20px;
	text-align: center;
	cursor: pointer;
	border-radius: 5px;
	box-sizing: border-box;

	color: ${(props) => props.theme.darkerGray};
	background-color: ${(props) => props.theme.white};
	border: 1px solid ${(props) => props.theme.lightGray};

	transition: color 0.2s ease-in, background-color 0.2s ease-in, border 0.2s ease-in;
	:hover,
	:active {
		color: ${(props) => props.theme.white};
		background-color: ${(props) => props.theme.primary};
		border: 1px solid ${(props) => props.theme.primary};
	}
	:disabled {
		cursor: not-allowed;
		color: ${(props) => props.theme.gray};
		border: 1px solid ${(props) => props.theme.lighterGray};
	}
`
