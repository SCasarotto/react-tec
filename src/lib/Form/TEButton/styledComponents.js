import styled from 'styled-components'

export const Button = styled.button`
	${(props) => {
		const { disabled, theme } = props

		return `
			position: relative;
			display: block;
			width: 100%;
			font-size: 18px;
			padding: 5px 20px;
			text-align: center;
			cursor: ${disabled ? 'not-allowed' : 'pointer'};
			border-radius: 5px;
            box-sizing: border-box;

			color: ${disabled ? theme.gray : theme.darkerGray};
			border: 1px solid ${disabled ? theme.lighterGray : theme.lightGray};

			transition: color 0.2s ease-in, background-color 0.2s ease-in, border 0.2s ease-in;
			:hover {
				color: ${theme.white};
				background-color: ${theme.primary};
				border: 1px solid ${theme.primary};
			}
			:active {
				color: ${theme.white};
				background-color: ${theme.primary};
				border: 1px solid ${theme.primary};
			}
		`
	}}
`
