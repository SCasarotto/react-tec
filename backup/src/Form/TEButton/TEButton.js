import React from 'react'

import { Button } from './styledComponents'

const TEButton = (props) => {
	const { className = '', type = 'button', ...rest } = props
	return <Button className={`TEButton ${className}`} type={type} {...rest} />
}

export default TEButton
