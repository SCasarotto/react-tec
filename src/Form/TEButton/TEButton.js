import React from 'react'

import { Button } from './styledComponents'

const TEButton = (props) => <Button {...props} />

TEButton.defaultProps = {
	type: 'button',
}

export default TEButton
