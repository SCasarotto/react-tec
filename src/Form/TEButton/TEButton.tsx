import React from 'react'

import { Button } from './styledComponents'

export interface TEButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {}
export const TEButton = React.forwardRef<HTMLButtonElement, TEButtonProps>((props, ref) => {
	const { className = '', type = 'button', ...rest } = props
	return <Button className={`TEButton ${className}`} type={type} ref={ref} {...rest} />
})
