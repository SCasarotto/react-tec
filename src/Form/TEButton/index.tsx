import React from 'react'

import { Button } from './styledComponents'

export interface TEButtonProps
	extends React.DetailedHTMLProps<
		React.ButtonHTMLAttributes<HTMLButtonElement>,
		HTMLButtonElement
	> {
	//TODO: Fix this. styled-components is weird with refs
	ref?: any
}
export const TEButton: React.FC<TEButtonProps> = (props) => {
	const { className = '', type = 'button', ...rest } = props
	return <Button className={`TEButton ${className}`} type={type} {...rest} />
}
