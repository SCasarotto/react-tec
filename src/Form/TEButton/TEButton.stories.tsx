import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TEButton, TEButtonProps } from './TEButton'

export default {
	title: 'Form/TEButton',
	component: TEButton,
	argTypes: {
		children: { control: 'text' },
		disabled: { control: 'boolean' },
	},
} as Meta

const Template: Story<TEButtonProps> = (args) => <TEButton {...args} />

export const Default = Template.bind({})
Default.args = {
	children: 'Default',
	onClick: () => alert('Clicked!'),
}
export const Disabled = Template.bind({})
Disabled.args = {
	children: 'Disabled',
	disabled: true,
}
