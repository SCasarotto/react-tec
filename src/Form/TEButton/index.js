import TEButton from './TEButton'

export default TEButton
