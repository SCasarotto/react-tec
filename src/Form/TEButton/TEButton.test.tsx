import React from 'react'
import { render } from '@testing-library/react'
import { TEButton } from './TEButton'

describe('TEButton', () => {
	it('matches snapshot', () => {
		const { asFragment } = render(<TEButton>My Test Button</TEButton>)
		expect(asFragment()).toMatchSnapshot()
	})
})
