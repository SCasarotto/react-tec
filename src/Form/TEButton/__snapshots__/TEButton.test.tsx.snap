// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEButton matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-bdnylx ccWmoN TEButton "
    type="button"
  >
    My Test Button
  </button>
</DocumentFragment>
`;
